File: CS/DXApplication2/MyUserSettings.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.Configuration;
using System.Drawing;
using System.Windows.Forms;

namespace DXApplication2 {
    public class MyUserSettings : ApplicationSettingsBase {
        [UserScopedSetting()]
        public BindingList<Host> Hosts {
            get {
                return ((BindingList<Host>)this["Hosts"]);
            }
            set { this["Hosts"] = value; }
        }
        //[UserScopedSetting()]
        //public BindingList<Rectangle> HostPositions {
        //    get { return ((BindingList<Rectangle>)this["HostPositions"]); }
        //    set { this["HostPositions"] = value; }
        //}
    }

    public class Host {
        public Host() {
        }

        public string HostID {
            get;
            set;
        }
        public Rectangle HostBounds {
            get;
            set;
        }
        public FormWindowState FormState {
            get;
            set;
        }
    }
}

File: VB/DXApplication2/MyUserSettings.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Configuration
Imports System.Drawing
Imports System.Windows.Forms

Namespace DXApplication2
    Public Class MyUserSettings
        Inherits ApplicationSettingsBase

        <UserScopedSetting()> _
        Public Property Hosts() As BindingList(Of Host)
            Get
                Return (DirectCast(Me("Hosts"), BindingList(Of Host)))
            End Get
            Set(ByVal value As BindingList(Of Host))
                Me("Hosts") = value
            End Set
        End Property
        '[UserScopedSetting()]
        'public BindingList<Rectangle> HostPositions {
        '    get { return ((BindingList<Rectangle>)this["HostPositions"]); }
        '    set { this["HostPositions"] = value; }
        '}
    End Class

    Public Class Host
        Public Sub New()
        End Sub

        Public Property HostID() As String
        Public Property HostBounds() As Rectangle
        Public Property FormState() As FormWindowState
    End Class
End Namespace